file-formats/nrob/nrob.json

```diff
@@ -0,0 +1,157 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://github.com/SAP/abap-file-formats/blob/main/file-formats/nrob/nrob.json",
+  "title": "NROB Object Type",
+  "description": "Object type NROB",
+  "type": "object",
+  "properties": {
+    "$schema": {
+      "title": "Schema",
+      "description": "Format version",
+      "type": "string"
+    },
+    "header": {
+      "title": "Header",
+      "description": "Header",
+      "type": "object",
+      "properties": {
+        "description": {
+          "title": "Description",
+          "description": "Description of the ABAP object",
+          "type": "string",
+          "maxLength": 60
+        },
+        "originalLanguage": {
+          "title": "Original Language",
+          "description": "Original language of the ABAP object",
+          "type": "string",
+          "maxLength": 2
+        },
+        "abapLanguageVersion": {
+          "title": "ABAP Language Version",
+          "description": "ABAP language version",
+          "type": "string",
+          "enum": [
+            "standard",
+            "keyUser",
+            "cloudDevelopment"
+          ],
+          "enumDescriptions": [
+            "Standard",
+            "ABAP for key user extensibility",
+            "ABAP cloud development"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "description",
+        "originalLanguage",
+        "abapLanguageVersion"
+      ]
+    },
+    "interval": {
+      "title": "Interval",
+      "description": "Interval",
+      "type": "object",
+      "properties": {
+        "numberLengthDomain": {
+          "title": "Number Length Domain",
+          "description": "Name of domain, which determines the length of the number range number. Define the amount of characters available for number range intervals. It must be of type NUMC or CHAR and have a field length of at least 1 and at most
+20.",
+          "type": "string",
+          "maxLength": 30
+        },
+        "percentage": {
+          "title": "Percent Warning",
+          "description": "Percentage of numbers remaining in a number range, upon reaching which in number assignment a warning is given. It must be between 0.1 and 99.9.\n\nExample: You have defined an interval from 1 to 1000. If you want to issue a
+ warning at the number 900, enter 10 (%) here.",
+          "type": "number",
+          "minimum": -9999.9,
```

Comment:
TYPE p LENGTH 3 DECIMALS 1

here also negative values are allowed by the schema. There is no better abap type which allows only positive double values. 
Solution 1: We check/verify in abap coding if the percentage is positive
Solution 2: We create another Abap Doc "annotation"

Reply:
In ABAP it has to be checked anyhow. 
Most probably it make sense to overrule the default values for min and max of a type. 

E.g., consider 100% would have been fine as well, then you also need an option to specify this.

Reply:
We will introduce some more abap doc "annotations".
$minimum:x
$maximum:x
$exclusiveMinimum:x
$exclusiveMaximum:x
$multipleOf:x